### DISCUSSION TITLE
v2.7 - Code Interpreter Exclusive

### DISCUSSION THREAD

**Discussion by JushBJJ** [JushBJJ/Mr.-Ranedeer-AI-Tutor#74] [Announcements]
# Mr. Ranedeer v2.7 - Code Interpreter Exclusive

Tokens: **5,560**

1. Re-arranged prompt to improve effectiveness
2. Added "Overall Rules to follow" header
3. Added a little personality to Mr. Ranedeer
4. Gave Mr. Ranedeer more freedom on what to say
5. Added `Prerequisite` and `Main Curriculum` examples as a strong guide
6. Removed `teach` function
7. Moved Mr. Ranedeer's thoughts onto the code environment
8. Removed token check & magic number as it is now no longer needed
9. Mr. Ranedeer will now wait for you response after it asks you a question
10. Reworked algorithm for `Curriculum` and `Lesson`
11. Added a standard for `Config Example`
12. Moved function rules to the bottom of the prompt
13. Recall personality and rules
14. Recall curriculum
15. Write instructions, emojis, and observations of the student
16. Ability to change the configuration automatically if needed
17. Ability to solve mathematical problems in python first in the /test command before showing them to the student.
18. Improve question accuracy by only giving one question at a time #40 
19. Create curriculum by uploading `.txt` files
20. Added configuration wizard #33

Mr. Ranedeer v2.7 Direct link: [https://chat.openai.com/share/53b38605-68e5-4922-81fe-e4588fb28d8a](https://chat.openai.com/share/53b38605-68e5-4922-81fe-e4588fb28d8a)

Configuration Wizard Direct link: [https://chat.openai.com/share/bb0d35d9-0239-492e-9ec2-49505aae202b](https://chat.openai.com/share/bb0d35d9-0239-492e-9ec2-49505aae202b)

<hr /><em>This discussion was created from the release <a href='https://github.com/JushBJJ/Mr.-Ranedeer-AI-Tutor/releases/tag/'>v2.7 - Code Interpreter Exclusive</a>.</em>

**Comment by A-F-V** [comment]
Thanks for a super interesting update! A question: why would I choose to update to v2.7 from 2.6? As a user of the GPT API, it makes it harder for me to incorporate into my use case and am wondering what improvements would warrant to change from my change?

For example, what is the motivation for saving to curriculum to a file? Does it actually fetch that file later? Is there a possibility to resume lessons with a file and code interpreter?
